pkg/kstatus/observe/aggregator/aggregator.go

```diff
@@ -0,0 +1,73 @@
+package aggregator
+
+import (
+	"sigs.k8s.io/cli-utils/pkg/kstatus/observe/event"
+	"sigs.k8s.io/cli-utils/pkg/kstatus/status"
+	"sigs.k8s.io/cli-utils/pkg/kstatus/wait"
+)
+
+// AllCurrentOrNotFoundStatusAggregator is an implementation of the StatusAggregator interface
+// that considers the aggregate status as Current whenever all resources have either the Current
+// or the NotFound status.
+// TODO: Treating resources that doesn't exist as Current is weird. But it kinda does
+// make sense when we track resources that are deleted/pruned. We should see if there is a better
+// way to handle this.
+type AllCurrentOrNotFoundStatusAggregator struct {
+	resourceCurrentStatus map[wait.ResourceIdentifier]status.Status
+}
+
+// NewAllCurrentOrNotFoundStatusAggregator returns a new StatusAggregator that will track
+// the status of all resources given by the provided identifiers.
+func NewAllCurrentOrNotFoundStatusAggregator(identifiers []wait.ResourceIdentifier) *AllCurrentOrNotFoundStatusAggregator {
+	aggregator := &AllCurrentOrNotFoundStatusAggregator{
+		resourceCurrentStatus: make(map[wait.ResourceIdentifier]status.Status),
+	}
+	for _, id := range identifiers {
```

Comment:
empty list not an error?

Reply:
The framework is set up to handle an empty slice of identifiers. Not clear where that is useful, but it will work and simply return that the aggregate status is Current.